src/d3d9/d3d9_common_texture.cpp

```diff
@@ -40,6 +40,8 @@ namespace dxvk {
 
     m_mapMode        = DetermineMapMode();
     m_shadow         = DetermineShadowState();
+    m_upgradedToD32f = ConvertFormatUnfixed(m_desc.Format).FormatColor != VK_FORMAT_D32_SFLOAT_S8_UINT &&
```

Comment:
Just set this after image creation and check if the VkFormat matches the desc format.

Reply:
If desc.format == D24 && image->info->format == VK_Format_d32

Reply:
well but if a driver doesn't support D16 and D24, that's not good enough.

Reply:
Sure.